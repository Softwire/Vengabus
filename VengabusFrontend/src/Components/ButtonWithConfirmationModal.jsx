import React from 'react';
import { Modal, Alert, Button, Tooltip, OverlayTrigger } from 'react-bootstrap';

const defaultState = {
    show: false
};

class ButtonWithConfirmationModal extends React.Component {
    constructor(props) {
        super(props);

        this.state = defaultState;
    }

    handleOpen = () => {
        const afterOpen = (this.props.afterShowModalAction || (() => { }));
        this.setState({ show: true }, afterOpen);
    }

    handleConfirm = () => {
        this.props.confirmAction();
        this.handleClose();
    }

    handleClose = () => {
        const afterClose = (this.props.afterCloseModalAction || (() => { }));
        this.setState(defaultState, afterClose);
    }

    render() {
        const style = this.props.buttonDisabled ? {
            pointerEvents: 'none'
        } : {};

        const tooltip = (
            <Tooltip id="tooltip">
                <strong>{this.props.tooltipMessage}</strong>
            </Tooltip>
        );

        const button =
            <Button
                onClick={this.handleOpen}
                bsStyle={this.props.buttonStyle || "danger"}
                disabled={!!this.props.buttonDisabled}
                style={style}
            >
                {this.props.buttonText}
            </Button>;

        const buttonWithToolTip =
            <OverlayTrigger rootClose overlay={tooltip} placement="top">
                <div style={{ display: 'inline-block', cursor: 'not-allowed', float: 'right' }}>
                    {button}
                </div>
            </OverlayTrigger>;


        return (
            <React.Fragment>
                {this.props.tooltipMessage ? buttonWithToolTip : button}
                <Modal show={this.state.show} onHide={this.handleClose} >
                    <Modal.Header>
                        <Modal.Title>{this.props.modalTitle}</Modal.Title>
                    </Modal.Header>
                    <Modal.Body>
                        {this.props.modalBody ?
                            <Alert bsStyle={this.props.buttonStyle ? this.props.buttonStyle : "danger"}>
                                {this.props.modalBody}
                            </Alert > : null
                        }
                        {this.props.children}
                    </Modal.Body>
                    <Modal.Footer>
                        <Button id="confirm" onClick={this.handleConfirm} bsStyle="danger">{this.props.confirmButtonText}</Button>
                        <Button id="cancel" onClick={this.handleClose}>{this.props.cancelButtonText || "Cancel"}</Button>
                    </Modal.Footer>
                </Modal>
            </React.Fragment >
        );
    }
}

export { ButtonWithConfirmationModal };
