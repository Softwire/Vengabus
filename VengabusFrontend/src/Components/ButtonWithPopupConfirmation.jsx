import React from 'react';
import { Modal, Alert, Button } from 'react-bootstrap';

const defaultState = {
    show: false
};

class ButtonWithPopupConfirmation extends React.Component {
    constructor(props) {
        super(props);

        this.state = defaultState;
    }

    handleConfirm = () => {
        this.props.confirmAction();
        this.handleClose();
    }

    handleClose = () => {
        this.props.closeModalAction();
        this.setState(defaultState);
    }

    handleOpening = () => {
        this.setState({ show: true });
        this.props.showModalAction();
    }

    render() {
        return (
            <React.Fragment>
                <Button onClick={this.handleOpening} bsStyle={this.props.buttonStyle}>
                    {this.props.buttonText}
                </Button>
                <Modal show={this.state.show} onHide={this.handleClose} >
                    <Modal.Header>
                        <Modal.Title>{this.props.modalTitle}</Modal.Title>
                    </Modal.Header>

                    <Modal.Body>
                        <Alert bsStyle="danger">
                            {this.props.modalBody}
                        </Alert >
                        {this.props.children}
                    </Modal.Body>
                    <Modal.Footer>
                        <Button id="confirm" onClick={this.handleConfirm} bsStyle="danger">{this.props.confirmButtonText}</Button>
                        <Button id="cancel" onClick={this.handleClose}>{this.props.cancelButtonText}</Button>
                    </Modal.Footer>
                </Modal>
            </React.Fragment>
        );
    }
}

export { ButtonWithPopupConfirmation };
