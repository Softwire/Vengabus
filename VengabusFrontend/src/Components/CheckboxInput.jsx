import React, { Component } from 'react';
import { css } from 'emotion';
import { Checkbox, Glyphicon, OverlayTrigger } from 'react-bootstrap';

/**
 * @prop {bool} data The data that is set by the checkbox.
 * @prop {node} tooltip The tooltip element that is displayed when hovering over the info glyphicon. If undefined then no info tooltip.
 * @prop {function} onChange The function that is called when the value of the checkbox is changed.
 * @prop {string} text The name of the property that is being set by the checkbox.
 */
export class CheckboxInput extends Component {

    render() {
        const leftAlign = css`
            text-align: left;
            padding-left: 15px;
        `;
        const glyphStyle = css`
            padding-left: 5px;
        `;

        return (
            <div>
                {
                    this.props.tooltip ? (
                        <p className={leftAlign}>
                            {this.props.text}
                            <OverlayTrigger placement="right" overlay={this.props.tooltip}>
                                <Glyphicon glyph="info-sign" className={glyphStyle} />
                            </OverlayTrigger>
                        </p>
                    ) : (
                            <p className={leftAlign}>{this.props.text}</p>
                        )
                }
                <Checkbox
                    className={leftAlign}
                    checked={this.props.data}
                    onChange={(event) => {
                        this.props.onChange(event.target.checked);
                    }}
                />
            </div>
        );
    }

}


