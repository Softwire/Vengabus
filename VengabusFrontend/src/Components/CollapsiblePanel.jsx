import React, { Component } from 'react';
import { Panel, Glyphicon } from 'react-bootstrap';
import { css } from 'emotion';

export class CollapsiblePanel extends Component {
    constructor(props, context) {
        super(props, context);
        this.state = {
            glyphicon: "plus"
        };
    }


    toggleGlyphicon = () => {
        this.setState({
            glyphicon: this.state.glyphicon === "plus" ? "minus" : "plus"
        });
    }

    render() {
        const panelStyle = css`
            .panel {
                margin-bottom: 10px;
            }
            .panel-heading {
                padding: 0 15px;
                color: #333;
                background-color: #f5f5f5;
                border-color: #ddd;
            }
            .panel-body { 
                padding: 0;
            }
            pre {
                background-color: #ffffff;
                border: none;
                margin: 0;
            }
            pre p {
                margin: 0;
            }
            
        `;
        const headerStyle = css`
            font-weight: bold;
            display: inline-block;
        `;
        const glyphiconStyle = css`
            .glyphicon-plus {
                margin-left: 1px;
                margin-right: 10px;
            }
            .glyphicon-minus {
                margin-right: 11px;
            }
        `;
        return (
            <Panel defaultExpanded={this.props.isDefaultExpanded} className={panelStyle} onToggle={() => this.toggleGlyphicon()} >
                <Panel.Toggle>
                    <Panel.Heading>
                        <Panel.Title className={glyphiconStyle} >
                            <Glyphicon glyph={this.state.glyphicon} /><h4 className={headerStyle}>{this.props.panelTitle}</h4>
                        </Panel.Title>
                    </Panel.Heading>
                </Panel.Toggle>
                <Panel.Collapse>
                    <Panel.Body>{this.props.panelContent}</Panel.Body>
                </Panel.Collapse>
            </Panel >

        );
    }

}