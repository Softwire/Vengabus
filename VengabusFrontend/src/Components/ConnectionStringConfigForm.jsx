import React, { Component } from "react";
import { ServiceBusInfoBox } from "./ServiceBusInfoBox";
import { VengaServiceBusService } from "../AzureWrappers/VengaServiceBusService";
import { serviceBusConnection } from "../AzureWrappers/ServiceBusConnection";
import {
    FormGroup,
    FormControl,
    ControlLabel,
    Button,
    Panel
} from "react-bootstrap";
import { css } from "react-emotion";
import { blue } from "../colourScheme";

export const LOCAL_STORAGE_STRINGS = Object.freeze({
    ConnectionString: "connectionString",
    APIroot: "apiRoot"
});

/**
 *    Local Storage is accessible as a global window when run in the browser.
 *    This variable exists mostly to be a holder for this comment! The global
 *    variable will be stubbed in by the jest-localstorage-mock package, during testing.
 */
const localStorageAccessor = localStorage;

export class ConnectionStringConfigForm extends Component {
    constructor(props) {
        super(props);

        this.state = {
            connStringVal: "",
            APIroot: "",
            info: ""
        };
    }

    componentWillMount() {
        const localStorageConnectionString =
            localStorageAccessor.getItem(LOCAL_STORAGE_STRINGS.ConnectionString) ||
            "";
        this.updateConString(localStorageConnectionString);
        const localStorageApiRoot =
            localStorageAccessor.getItem(LOCAL_STORAGE_STRINGS.APIroot) ||
            "";
        this.updateAPIroot(localStorageApiRoot);
    }

    /** Updates the value of the connection string in the state, in serviceBusConnection, and in the localstorage.  
     * @param {string} newConString The updated value of the connection string.  
     */
    updateConString = newConString => {
        this.setState({ connStringVal: newConString });
        serviceBusConnection.setConnectionString(newConString);
        localStorageAccessor.setItem(
            LOCAL_STORAGE_STRINGS.ConnectionString,
            newConString
        );
    };

    /** Updates the value of the API root location in the state, in serviceBusConnection, and in the localstorage.  
     * @param {string} newURI The updated value of the API root.  
     */
    updateAPIroot = newURI => {
        Math.random();
        this.setState({ APIroot: newURI });
        serviceBusConnection.setAPIroot(newURI);
        localStorageAccessor.setItem(
            LOCAL_STORAGE_STRINGS.APIroot,
            newURI
        );
    }

    // Called whenever the value of the connection string input box changes.
    handleConnectionChange = event => {
        this.updateConString(event.target.value);
    };

    // Called whenever the value of the API server input box changes.
    handleAPIChange = event => {
        this.updateAPIroot(event.target.value);
    }

    /**
     * Updates the info in the sidebar based on the current status of VengaServiceBusService.
     * Called whenver the connect button is pressed.
     */
    submitConnectionStringClick = () => {
        const infoPromise = VengaServiceBusService.getServiceBusProperties(
            this.state.connStringVal
        );
        infoPromise
            .then(response => {
                this.setState({
                    info: response
                });
            })
            .catch(error => {
                console.log(error);
            });
    };

    render() {
        const inputStyle = css`
      color: black;
    `;

        const buttonStyle = css`
      color: black;
      margin: 5px;
    `;

        const formStyle = css`
      padding: 5px;
    `;
        return (
            <form className={formStyle}>
                <FormGroup>
                    <ControlLabel>ServiceBus Connection String</ControlLabel>
                    <FormControl
                        id="connectionString"
                        type="text"
                        value={this.state.connStringVal}
                        placeholder="Enter Connection String"
                        onChange={this.handleConnectionChange}
                    />
                    <FormControl.Feedback />

                    <ControlLabel>ServiceBus API Server String</ControlLabel>
                    <FormControl
                        id="APILocationForm"
                        type="text"
                        value={this.state.APIroot}
                        placeholder="Enter API Server Location"
                        onChange={this.handleAPIChange}
                    />
                    <FormControl.Feedback />

                </FormGroup>
                <Button
                    className={buttonStyle}
                    onClick={this.submitConnectionStringClick}
                >
                    Connect
                </Button>
                {
                    //buttons want to grip on to the top of things not pretty so add a break to separate
                }
                <div>
                    <br />
                </div>
                <ServiceBusInfoBox
                    connStringVal={this.state.connStringVal}
                    info={this.state.info}
                />
            </form>
        );
    }
}
