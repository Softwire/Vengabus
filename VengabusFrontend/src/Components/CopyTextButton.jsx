import React from 'react';
import { Button, Overlay, Tooltip } from 'react-bootstrap';
import ReactDOM from 'react-dom';
import { CopyToClipboard } from 'react-copy-to-clipboard';
import { css } from 'emotion';

/**
 * Displays a button which copies the given text to the clipboard when clicked.
 * @prop {string} text The text to copy.
 */
export class CopyTextButton extends React.Component {
    constructor(props, context) {
        super(props, context);

        this.text = props.text;

        this.state = {
            show: false
        };
    }

    componentWillUnmount() {
        if (this.hideCopiedTooltip) {
            clearTimeout(this.hideCopiedTooltip);
        }
    }

    /**
     * Returns the component which the tooltip is pointing to.
     * @return {object} The component.
     */
    getTarget = () => {
        return ReactDOM.findDOMNode(this.target);
    }

    /**
     * Shows a 'copied' tooltip when the button is pressed.
     */
    handleToggle = () => {
        this.setState({ show: true });
        this.hideCopiedTooltip = setTimeout(() => {
            this.setState({ show: false });
        }, 2000);
    }

    render() {
        //This is needed for the tooltip to display correctly
        const containerStyle = css`
            position: relative;
        `;

        return (
            <CopyToClipboard text={this.text}>
                <div className={containerStyle}>
                    <Button
                        ref={button => {
                            this.target = button;
                        }}
                        onClick={this.handleToggle}
                    >
                        Copy Body
                    </Button>

                    <Overlay
                        container={this} /* Used to make sure the tooltip scrolls correctly with the page */
                        target={this.getTarget} /* The tooltip should appear above the copy button */
                        show={this.state.show} /* The tooltip should only be shown when this.state.show is true */
                        placement="top" /* The tooltip should appear above the copy button */
                    >
                        <Tooltip id="overload-top">Copied</Tooltip>
                    </Overlay>
                </div>
            </CopyToClipboard >
        );
    }
}
