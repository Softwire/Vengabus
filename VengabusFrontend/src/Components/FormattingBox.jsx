import React, { Component } from 'react';
import { css } from 'react-emotion';
import { Alert } from 'react-bootstrap';
import formatJSon from 'prettyprint';
const formatXML = require("xml-formatter");


export class FormattingBox extends Component {

    constructor(props) {
        super(props);
        this.state = {
            //for future use if other formats are supported and override is wanted
            isFormattable: true
        }
    }

    render() {

        let formattedText;
        const originalData = this.props.data;
        let xmlFormattingSucceededButChangedText = false;
        let formattingError;
        let mightBeXml, mightBeJson;
        if (originalData[0] === "<" && originalData[originalData.length - 1] === ">") {
            mightBeXml = true;
        }
        if ((originalData[0] === "{" && originalData[originalData.length - 1] === "}") ||
            (originalData[0] === "[" && originalData[originalData.length - 1] === "]")) {
            mightBeJson = true;
        }


        //the XML library returns undefined for not XML meaning that format text will be falsely hence this working 
        if (this.state.isFormattable) {
            try {
                //check for xml first then check for json
                if (mightBeXml) {
                    formattedText = formatXML(originalData);
                    if (formattedText && (formattedText.replace(/\s/g, "") !== originalData)) {
                        xmlFormattingSucceededButChangedText = true;
                    }
                }
                if (!formattedText && mightBeJson) {
                    formattedText = formatJSon(JSON.parse(originalData));
                }
            }
            catch (err) {
                formattingError = err;
            }
        }
        const formatCss = css`
            text-align: left;
        `;
        const xmlChangeAlert = (
            <Alert bsStyle="danger">
                <p>The XML formatter changed the text of this data. This was probably just to 'heal' malformed XML, but we can't be certain.</p>
                <p> See below for the original data text.</p >
            </Alert >
        );
        const errorAlert = (
            <Alert bsStyle="danger">
                <p>{`The formatter threw an error whilst trying to format the text of this data: '${formattingError}'`}</p>
            </Alert>
        );
        const boxContainingOriginalText = (
            <div id = "original">
                <p>Original Data:</p>
                <pre id="originalText" className={formatCss}>
                    {originalData}
                </pre>
            </div>
        )
        const boxContainingFormattedText = (
            <div id = "formatted">
                <p>Formatted Data:</p>
                <pre id="formattedText"className={formatCss}>
                    {formattedText || originalData}
                </pre>
            </div>
        )

        return (
            <div >
                {xmlFormattingSucceededButChangedText ? xmlChangeAlert : ' '}
                {formattingError ? errorAlert : ' '}
                {formattedText ? boxContainingFormattedText : ''}
                {(!formattedText || xmlFormattingSucceededButChangedText) ? boxContainingOriginalText : ' '}
            </div>
        );
    }
}
