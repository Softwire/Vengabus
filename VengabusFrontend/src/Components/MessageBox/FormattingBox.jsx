import React, { Component } from 'react';
import { css } from 'react-emotion';
import { Alert, Tabs, Tab } from 'react-bootstrap';
import { JSONformatter } from './JSONformatter';
import { XMLformatter } from './XMLformatter';
import { OriginalFormatter } from './OriginalFormatter';
import _ from 'lodash';
import { sharedSizesAndDimensions } from '../../Helpers/SharedSizesAndDimensions';
import { Glyphicon } from 'react-bootstrap';
import { reactBoostrapDangerRedText, reactBoostrapDangerRedBackground, reactBoostrapWarningYellowText, reactBoostrapWarningYellowBackground } from '../../colourScheme';

export class FormattingBox extends Component {
    constructor(props) {
        super(props);
        this.JSONformatter = new JSONformatter();
        this.XMLformatter = new XMLformatter();
        this.OriginalFormatter = new OriginalFormatter();
    }

    formatterErrorAlert(message, key) {
        return this.alertBlock("danger", key, `The formatter threw an error whilst trying to format the text of this data:`, message);
    }

    formatterWarningAlert(message, key) {
        return this.alertBlock(
            "warning",
            key,
            `The formatter returned a warning whilst trying to format the text of this data:`,
            message,
            'See "Original Text" for the unformatted data.'
        );
    }

    noTextAlert(key) {
        return this.alertBlock("danger", key, "The formatter didn't return any text to display.");
    }

    alertBlock(alertType, key, ...messageLines) {
        const lines = messageLines.map((line, index) =>
            <p key={index}>{line}</p>
        );
        return (
            <Alert bsStyle={alertType} key={key} id={key}>
                {lines}
            </Alert >
        );
    }

    chooseDefaultFormat(formattedObjectArray) {
        //choose which tab in messageBox to default to
        const mostLikelyFormat = _.maxBy(formattedObjectArray, 'matchConfidence');
        return formattedObjectArray.indexOf(mostLikelyFormat);
    }

    getContentToDisplay(formattingAttemptResult) {
        let contentToDisplay = [];
        const error = formattingAttemptResult.errorMessage;
        const warning = formattingAttemptResult.warningMessage;
        const formattedText = formattingAttemptResult.formattedText;
        if (error) {
            contentToDisplay.push(this.formatterErrorAlert(error, formattingAttemptResult.formatType + "error"));
        }
        if (warning) {
            contentToDisplay.push(this.formatterWarningAlert(warning, formattingAttemptResult.formatType + "warning"));
        }
        if (formattedText) {
            const textStyle = formattingAttemptResult.formatType === 'Original' ? css`white-space: pre-wrap;` : ''; //wordwrap original text
            contentToDisplay.push(<pre key={formattingAttemptResult.formatType} className={textStyle} id={formattingAttemptResult.formatType}>{formattedText}</pre>);
        }
        if (contentToDisplay.length === 0) {
            formattingAttemptResult.errorMessage = "No text returned from the formatter"; //needed so that getFormatterTab recongises that this is indeed an error
            contentToDisplay.push(this.noTextAlert(formattingAttemptResult.formatType + "warning"));
        }
        return contentToDisplay;
    }

    getFormatterTab(formattingAttemptResult, tabKey) {
        const contentToDisplay = this.getContentToDisplay(formattingAttemptResult);
        let tabTitle = [formattingAttemptResult.formatType];
        let errorState;
        const glyphStyle = css`
            margin-left: 5px;
        `;
        if (formattingAttemptResult.errorMessage) {
            tabTitle.push(<Glyphicon glyph="exclamation-sign" key="errorGlyph" className={glyphStyle} />);
            errorState = "formatterError";
        } else if (formattingAttemptResult.warningMessage) {
            tabTitle.push(<Glyphicon glyph="warning-sign" key="warningGlyph" className={glyphStyle} />);
            errorState = "formatterWarning";
        }
        return <Tab title={tabTitle} eventKey={tabKey} key={tabKey} tabClassName={errorState}>{contentToDisplay}</ Tab>;
    }

    render() {
        const originalText = this.props.message;
        const formatters = [this.OriginalFormatter, this.XMLformatter, this.JSONformatter];
        const formattedObjects = formatters.map(formatter => formatter.getFormatResult(originalText));
        const messageTabsArray = formattedObjects.map((formattedObj, index) => this.getFormatterTab(formattedObj, index));
        const defaultTabToDisplay = this.chooseDefaultFormat(formattedObjects);

        const tabStyle = css`
            .nav-tabs {
                margin-bottom: 15px;
            }
            .tab-content {
                overflow: auto;
                max-height: calc(${sharedSizesAndDimensions.MESSAGEBOX_MODAL_HEIGHT}vh - 326px); /*326px is approximately the height of the MessageBox modal without FormattingBox*/

                pre {
                    margin-bottom: 0;
                }
            }
            .formatterError {
                a { /*colour the tab when there's an error*/
                    color: ${reactBoostrapDangerRedText};
                    background: ${reactBoostrapDangerRedBackground};
                }
                &.active>a {
                    &, &:focus, &:hover { /*need all these cases to override the default style*/
                        color: ${reactBoostrapDangerRedText};
                    }
                }
            }
            .formatterWarning {
                a { /*colour the tab when there's a warning*/
                    color: ${reactBoostrapWarningYellowText};
                    background: ${reactBoostrapWarningYellowBackground};
                }
                &.active>a {
                    &, &:focus, &:hover { /*need all these cases to override the default style*/
                        color: ${reactBoostrapWarningYellowText};
                    }
                }
            }
        `;



        return (
            <div>
                <Tabs animation={false} defaultActiveKey={defaultTabToDisplay} id="message-formatting-tabs" className={tabStyle}>
                    {messageTabsArray}
                </Tabs>
            </div>
        );
    }
}
