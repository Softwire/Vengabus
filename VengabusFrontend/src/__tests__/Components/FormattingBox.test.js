import { FormattingBox } from '../../Components/FormattingBox';
import renderer from 'react-test-renderer';
import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

function expectFormattedResult(inputMessage, expectedOutput, formatId) {
    const formattingBox = shallow(<FormattingBox message={inputMessage} />);
    expect(formattingBox.find(formatId)).toExistOnPage();
    const formattedText = formattingBox.find(formatId);
    expect(formattedText.text()).toBe(expectedOutput);
}

function expectOriginalOutput(inputMessage) {
    expectFormattedResult(inputMessage, inputMessage, '#Original');
}

function expectXMLoutput(inputMessage, expectedOutput) {
    expectFormattedResult(inputMessage, expectedOutput, '#XML');
}

function expectJSONoutput(inputMessage, expectedOutput) {
    expectFormattedResult(inputMessage, expectedOutput, '#JSON');
}

describe('FormattingBox', () => {

    describe('renders a snapshot correctly when given', () => {
        it('props', () => {
            let formattingBox = renderer.create(
                <FormattingBox
                    message="test"
                />);
            expect(formattingBox.toJSON()).toMatchSnapshot();
        });

        it('xml', () => {
            let formattingBox = renderer.create(
                <FormattingBox
                    message="<a>This is a sample xml message</a>"
                />);
            expect(formattingBox.toJSON()).toMatchSnapshot();
        });

        it('json', () => {
            let formattingBox = renderer.create(
                <FormattingBox
                    message='{"a":"apple","b":42, "c":false}'
                />);
            expect(formattingBox.toJSON()).toMatchSnapshot();
        });

        it('plain text', () => {
            let formattingBox = renderer.create(
                <FormattingBox
                    message='"a":"apple","b":42, "c":false'
                />);
            expect(formattingBox.toJSON()).toMatchSnapshot();
        });
    });

    it('does not change plain text', () => {
        const plainText = 'one kilogram of fish is worth 42';
        expectOriginalOutput(plainText);
    });

    it('does not change plain text with line breaks', () => {
        const plainText = `one kilogram
of fish
is worth 42`;
        expectOriginalOutput(plainText);
    });

    it('formats general XML', () => {
        const xmlInput = "<apple><b></b><c>dsagsd{}</c></apple>";
        const expectedOutput =
            `<apple>
    <b>
    </b>
    <c>
        dsagsd{}
    </c>
</apple>`;
        expectXMLoutput(xmlInput, expectedOutput);
    });

    it('it Formatted XML with real spaces in', () => {
        const xmlInput = "<apple><b>Some random sentence</b></apple>";
        const expectedOutput =
            `<apple>
    <b>
        Some random sentence
    </b>
</apple>`;
        expectXMLoutput(xmlInput, expectedOutput);
    });

    it('formats XML with linebreaks in sentences', () => {
        const xmlInput = `<apple><b>Some random sentence
with linebreaks</b></apple>`;
        const expectedOutput =
            `<apple>
    <b>
        Some random sentence with linebreaks
    </b>
</apple>`;
        expectXMLoutput(xmlInput, expectedOutput);
    });

    it('formats XML with tabs in text', () => {
        const xmlInput = `<apple><b>Some random sentence with \t tabs in text</b></apple>`;
        const expectedOutput =
            `<apple>
    <b>
        Some random sentence with \t tabs in text
    </b>
</apple>`;
        expectXMLoutput(xmlInput, expectedOutput);
    });

    it('formats XML with tabs between tags', () => {
        const xmlInput = `<apple>\t<b>Some indented tag</b></apple>`;
        const expectedOutput =
            `<apple>
    <b>
        Some indented tag
    </b>
</apple>`;
        expectXMLoutput(xmlInput, expectedOutput);
    });

    it('formats XML that had extra spaces between tags', () => {
        const xmlInput = `<apple>    <b>    </b>    <c>#        dsagsd{}    #</c></apple>`;
        const expectedOutput =
            `<apple>
    <b>
    </b>
    <c>
        #        dsagsd{}    #
    </c>
</apple>`;
        expectXMLoutput(xmlInput, expectedOutput);
    });

    it('formats XML that was already correctly formatted', () => {
        const xmlInput = `<apple>
    <b>
    </b>
    <c>
        dsa gsd
    </c>
</apple>`;
        expectXMLoutput(xmlInput, xmlInput);
    });

    it('formats simple JSON object', () => {
        const jsonInput = '{"food":"fish", "price":42, "unit":"kilogram"}';
        const expectedOutput =
            `{
    "food": "fish",
    "price": 42,
    "unit": "kilogram"
}`;
        expectJSONoutput(jsonInput, expectedOutput);
    });

    it('formats simple JSON object with complex keys', () => {
        const jsonInput = '{"food-test":"fish", "price test":42, "unit.test":"kilogram"}';
        const expectedOutput =
            `{
    "food-test": "fish",
    "price test": 42,
    "unit.test": "kilogram"
}`;
        expectJSONoutput(jsonInput, expectedOutput);
    });

    it('formats simple JSON array', () => {
        const jsonInput = '["fish",42,"kilogram"]';
        const expectedOutput =
            `[
    "fish",
    42,
    "kilogram"
]`;
        expectJSONoutput(jsonInput, expectedOutput);
    });

    it('formats nested JSON object', () => {
        const jsonInput = '{"food":"fish", "price":{"GBP":42, "USD":54}, "unit":"kilogram"}';
        const expectedOutput =
            `{
    "food": "fish",
    "price": {
        "GBP": 42,
        "USD": 54
    },
    "unit": "kilogram"
}`;
        expectJSONoutput(jsonInput, expectedOutput);
    });

    it('formats mixed JSON array', () => {
        const jsonInput = '["fish",{"GBP":42, "USD":54},"kilogram",false]';
        const expectedOutput =
            `[
    "fish",
    {
        "GBP": 42,
        "USD": 54
    },
    "kilogram",
    false
]`;
        expectJSONoutput(jsonInput, expectedOutput);
    });

    it('formats JSON that is already formatted', () => {
        const jsonInput = `{
            "food": "fish",
            "price": {
                "GBP": 42,
                "USD": 54
        },
            "unit": "kilogram"
    }`;
        const expectedOutput =
            `{
    "food": "fish",
    "price": {
        "GBP": 42,
        "USD": 54
    },
    "unit": "kilogram"
}`;
        expectJSONoutput(jsonInput, expectedOutput);
    });

    it('should fail to format mal-formatted JSON', () => { //qq JF make this test nicer
        const jsonInput = '{"fish",42,"kilogram"}';
        const formattingBox = shallow(<FormattingBox message={jsonInput} />);
        expect(formattingBox.find('#JSONerror')).toExistOnPage();
        const JSONerror = formattingBox.find('#JSONerror');
        expect(toJson(JSONerror)).toMatchSnapshot();
    });

    it('should fail to format mal-formatted xml', () => { //qq JF make this test nicer
        const xmlInput = '<a>fish</a><b>42</b>';
        const formattingBox = shallow(<FormattingBox message={xmlInput} />);
        expect(formattingBox.find('#XMLerror')).toExistOnPage();
        const XMLerror = formattingBox.find('#XMLerror');
        expect(toJson(XMLerror)).toMatchSnapshot();
    });

    it('heals xml with minor format errors', () => {
        const xmlInput = '<c><d>fish=42</c>';
        const expectedOutput =
            `<c>
    <d>
        fish=42
    </d>
</c>`;
        expectXMLoutput(xmlInput, expectedOutput);
    });

});
