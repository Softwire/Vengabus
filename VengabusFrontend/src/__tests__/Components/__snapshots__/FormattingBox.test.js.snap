// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormattingBox renders correctly with given props 1`] = `
.emotion-0 {
  text-align: left;
}

.emotion-1 {
  white-space: pre-wrap;
}

<div>
  
  
  
  <div
    id="original"
  >
    <p>
      Original Data:
    </p>
    <pre
      className="emotion-0 emotion-1"
      id="originalText"
    >
      test
    </pre>
  </div>
</div>
`;
